addons/mail/static/src/discuss/core/common/store_service_patch.js

```diff
@@ -3,6 +3,9 @@ import { patch } from "@web/core/utils/patch";
 
 /** @type {import("models").Store} */
 const storeServicePatch = {
+    get onlineMemberStatuses() {
+        return ["away", "bot", "online"];
+    },
```

Comment:
It's annoying to maintain these lists. Whenever we add / remove / adjust values, we have to maintain them, which sucks.

It would have been best to have:
- no im_status or .includes("offline") => user is offline category
- else => online

For member list categories, the case about Whatsapp user, this could be implemented by the component itself: it picks the most specific category elligible to the member, so that it can be a whatsapp user but cannot be in online at the same time.

Reply:
But there was `im_partner` aswell, which needed to placed inside Offline.

Moreover it's better to be specific what we need in the Online category, as that's the one which most people pay attention. Rest other statuses could just be put under offline unless you don't want it to.

It has chances of breaking in future as it was right now. Someone just tweaks the status for their specific needs which doesn't contain `offline` in it. Then it would be put inside the Online. While it shouldn't have been.

Reply:
Best would be to avoid combination of status, so im_status never change, and we just have the "on leave" on a separate field to combine into specific rendering if necessary.

Reply:
But for the fix I prefer if we just merge this PR, we can clean later.

Reply:
> But there was `im_partner` aswell, which needed to placed inside Offline.

`im_partner` yes also included in offline condition

> Moreover it's better to be specific what we need in the Online category, as that's the one which most people pay attention. Rest other statuses could just be put under offline unless you don't want it to.

The category "offline" is fully well defined: contains `offline`, `im_partner`, and no status
The rest either specifically defines a category, otherwise this is necessarily in "Online".

> It has chances of breaking in future as it was right now. Someone just tweaks the status for their specific needs which doesn't contain `offline` in it. Then it would be put inside the Online. While it shouldn't have been.

People tweaking new `im_status` that is "offline" should not happen. The listing of "offline" is totally under control of core code of `mail`. `leave_offline` is poor design that we keep support for the time being, but as @seb-odoo says this should ideally be in its own field / data and combine nicely with `im_status`.

What I don't like with enumerating all online and offline by hand is that future PRs to add new IM status (busy, custom) each have to remember to adapt this list. And we will definitely forget it and we'll need to run behind stupid bug fixes due to code that require more maintenance than necessary.